core/src/test/scala/com/github/mkroli/dns4s/section/resource/CAAResourceSpec.scala

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2019 Bilal Fazlani
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.mkroli.dns4s.section.resource
+
+import com.github.mkroli.dns4s.section.ResourceRecord
+import com.github.mkroli.dns4s.section.resource.CAAResource.{
+  IODEFResource,
+  IssueResource,
+  IssueWildResource,
+  CustomCAAResource
+}
+import com.github.mkroli.dns4s.{MessageBuffer, bytes}
+import org.scalatest.FunSpec
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+
+class CAAResourceSpec extends FunSpec with ScalaCheckDrivenPropertyChecks {
+  describe("CustomCAAResource") {
+    describe("validation") {
+      describe("tag") {
+        it("should fail if it is empty") {
+          intercept[IllegalArgumentException] {
+            CustomCAAResource("", "someValue".getBytes, Byte.MinValue)
+          }
+        }
+        it("should not fail for a valid value") {
+          CustomCAAResource("someTag", "someValue".getBytes, Byte.MinValue)
+        }
+      }
+    }
+    describe("encoding/decoding") {
+      it("decode(encode(resource)) should be the same as resource") {
+        val expectedResource =
+          CustomCAAResource("someTag", "someValue".getBytes, 1.toByte)
+        val encoded = expectedResource(MessageBuffer()).flipped()
+        val actualResource = CAAResource(encoded, encoded.remaining())
+        assert(expectedResource === actualResource)
+      }
```

Comment:
Can we add one assertion to test the deep equals?
```scala
assert(expectedResource.copy(valueBytes = expectedResource.valueBytes.map(b => (~b).toByte)) !== actualResource)
```

Reply:
Done 👍